flow/aggregates.py

```diff
@@ -354,7 +353,39 @@ def _register_aggregates(self):
         """If the instance of this class is called then we will
         generate aggregates and store them in ``self._aggregates``.
         """
-        self._aggregates, self._aggregate_ids = self._make_aggregates()
+        # Since we should always clear the aggregates before registering
+        # aggregates because we append the aggregates directly.
+        self._aggregates.clear()
+        self._aggregate_ids.clear()
+        aggregated_jobs = self._make_aggregates(self._project)
```

Comment:
We should pass in the project to this function and the `_create_nested_aggregate_list` function, or create the `_validate_and_filter_job` function in this function and pass that to `_create_nested_aggregate_list`. Either is fine with me. It is up to your preference.